test/gitea_test.go

```diff
@@ -637,6 +636,8 @@ func TestGiteaParamsOnRepoCR(t *testing.T) {
 	repo, err := topts.ParamsRun.Clients.PipelineAsCode.PipelinesascodeV1alpha1().Repositories(topts.TargetNS).Get(context.Background(), topts.TargetNS, metav1.GetOptions{})
 	assert.NilError(t, err)
 
+	// Checking repo status length to avoid index out of range [0] with length 0 issue
+	assert.Assert(t, len(repo.Status) != 0)
```

Comment:
@chmouel handled this way now so that it will fail here if status len is 0 
```
--- FAIL: TestGiteaParamsOnRepoCR (21.14s)
    gitea_test.go:645: assertion failed: len(repo.Status) is 0

```

Reply:
why would you think repo.Status is 0, it means the controller didn't update it properly so something went wrong i wonder what cause this flakyness
set as failure is the right way to do this.... it would fail as much as before, just differently, until we find the flakyness

Reply:
I thought something went wrong and repo status is empty 
based on this error

```
-- FAIL: TestGiteaParamsOnRepoCR (8.14s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 848 [running]:
testing.tRunner.func1.2({0x2cc6f60, 0xc000063d58})
	/opt/hostedtoolcache/go/1.19.12/x64/src/testing/testing.go:1396 +0x372
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.19.12/x64/src/testing/testing.go:1399 +0x5f0
panic({0x2cc6f60, 0xc000063d58})
	/opt/hostedtoolcache/go/1.19.12/x64/src/runtime/panic.go:890 +0x262
github.com/openshift-pipelines/pipelines-as-code/test.TestGiteaParamsOnRepoCR(0x0?)
	/home/runner/work/pipelines-as-code/pipelines-as-code/test/gitea_test.go:645 +0xf09
testing.tRunner(0xc0003d44e0, 0x2f32ff8)
```

But not sure what is causing that :thinking:

Reply:
True

Did not get the hint on why its failed

+1 for debug mode

> if we get this error on CI this may happen in customer installation 

Yes we might face

Reply:
So shall we wait for another fail and try to see the logs again instead of merging this PR now ?

Reply:
the assert still fails so it's fine to merge this...

we still need to go over the root cause of the issue...

Reply:
Sure
next time when assert fails I will try to look ate logs and see the reason